pkg/daemon/ceph/client/mirror.go

```diff
@@ -136,7 +136,7 @@ func enablePoolMirroring(context *clusterd.Context, clusterInfo *ClusterInfo, po
 		return errors.Wrapf(err, "failed to get mirroring info for the pool %q", pool.Name)
 	}
 
-	if pool.Mirroring.Mode == mirrorInfo.Mode {
+	if mirrorInfo.Mode != "" && mirrorInfo.Mode != "disabled" && pool.Mirroring.Mode != mirrorInfo.Mode {
```

Comment:
Another way to think about this check is by flipping the conditional. I'm not suggesting to change the code to this approach, I'm just trying to make sure we understand the condition.
```go
if mirrorInfo.Mode == "" || mirrorInfo.Mode == "disabled" || pool.Mirroring.Mode == mirrorInfo.Mode {
   // run the command to enable the pool mirroring
}
```
I'm not sure this makes sense. Isn't there ever a need to enable the mirroring when `pool.Mirroring.Mode != mirrorInfo.Mode`?

Reply:
We do supported changing mirroring mode but with other and `&&` condition will be good

Reply:
https://github.com/rook/rook/pull/15592 PR where I did testing

Reply:
So our goal is skip or not let the blockpool controller to override the mirroring on the pool which is done by rns controller

Reply:
@travisn, with your suggestion, are we saying even for pool level mirroring(not the namespace mirror), user will need to create radosnamespace CR?

Reply:
Yes, since we have a conflict, the mirroring setting should be controlled in a single place. It seems the radosnamespace controller is the place to own it, and we would require the rns CR to be created for the implicit namespace. This likely means the mirroring settings on the blockpool become obsolete? This needs more design discussion though.

Reply:
@Madhu-1 @subhamkrai What is the sequence expected for enabling mirroring in the pool, the implicit namespace, and other namespaces? It will really help to see it written down since I'm still not clear on it after you've attempted to talk me through it.

Reply:
I missed this ping. @subhamkrai it would be helpful to write down the sequence instead of upstream doc on how mirroring is enabled by the ocs-operator

* Mirroring is enabled on the pool level with init-only
* RNS is create for the implicit NS with mirroring mode set to `image`

@travisn The problem is only for implicit namespace not for other namespaces